command/ca/provisioner/webhook/webhook.go

```diff
@@ -0,0 +1,126 @@
+		Subcommands: cli.Commands{
+			addCommand(),
+			updateCommand(),
+			removeCommand(),
+		},
+		Description: `**step ca provisioner webhook** command group provides facilities for managing the webhooks attached to a provisioner
+
+Administrators can attach webhooks to provisioners to retrieve additional data that will be available when rendering certificate templates.
+Webhooks can also be used to disallow signing certificates for unknown entities.
+
+Any data returned from the webhook server will be added to the template context under the path "Webhooks.<name>".
+Implementations of webhook servers must conform to the step-ca documentation at https://smallstep.com/docs/step-ca/templates for parsing and verifying request bodies and forming valid response bodies.
+
+## EXAMPLES
+
+Add a new webhook to a provisioner:
+'''
+step ca provisioner webhook add my_webhook --provisioner my_provisioner --url https://example.com
+'''
+
+Change a webhook's url:
+'''
+step ca provisioner webhook update my_webhook --provisioner my_provisioner --url https://example.com
+'''
+
+Remove a webhook:
+'''
+step ca provisioner webhook remove my_webhook --provisioner my_provisioner
+'''
+		`,
+	}
+}
+
+var (
+	provisionerFlag = cli.StringFlag{
+		Name:     "provisioner",
+		Usage:    `The name of the provisioner the webhook is attached to.`,
+		Required: true,
+	}
+	urlFlag = cli.StringFlag{
+		Name:  "url",
+		Usage: `The url of the webhook server.`,
+	}
+	kindFlag = cli.StringFlag{
+		Name:  "kind",
+		Usage: `The kind of webhook. Default is ENRICHING.`,
+	}
+	bearerTokenFlag = cli.StringFlag{
+		Name:  "bearer-token",
+		Usage: `The token to be set in the Authorization header of the request to the webhook server.`,
+	}
+	basicAuthUsernameFlag = cli.StringFlag{
+		Name:  "basic-auth-username",
+		Usage: `The username portion of the Authorization header of the request to the webhook server when using basic authentication.`,
+	}
+	basicAuthPasswordFlag = cli.StringFlag{
+		Name:  "basic-auth-password",
+		Usage: `The password porition of the Authorization header of the request to the webhook server when using basic authentication.`,
+	}
+	disableTLSClientAuthFlag = cli.BoolFlag{
+		Name:  "disable-tls-client-auth",
+		Usage: `The CA will not send a client certificate when requested by the webhook server.`,
+	}
+	certTypeFlag = cli.StringFlag{
+		Name:  "cert-type",
+		Usage: `Whether to call this webhook when signing X509 certificates, SSH certificates, or ALL certificates. Default is ALL.`,
+	}
+)
+
+type crudClient interface {
+	GetProvisioner(...ca.ProvisionerOption) (*linkedca.Provisioner, error)
+	CreateProvisionerWebhook(provisionerName string, wh *linkedca.Webhook) (*linkedca.Webhook, error)
+	UpdateProvisionerWebhook(provisionerName string, wh *linkedca.Webhook) (*linkedca.Webhook, error)
+	DeleteProvisionerWebhook(provisionerName string, webhookName string) error
+}
+
+func newCRUDClient(cliCtx *cli.Context, cfgFile string) (crudClient, error) {
```

Comment:
Any reason no to export the CRUD client in ca/provisioner? I'm assuming the `newCRUDClient` code is the same.

Reply:
This one returns an error if the admin API is not enabled.